### PR TITLE
Return from RunWithCancel if cmd is kapp

### LINKED ISSUE #1627: kctrl package release  does not work as expected inside a kubernetes pod
**What steps did you take:**

1. Added `kctrl` as binary to the Dockerfile. 
2. Created a container image. 
3. Deployed a pod with the above image.
4. Ssh into the pod.
5. Ran `kctrl pkg release` inside the k8s pod.

**What happened:**
When I am trying to run `kctrl pkg release` inside the k8s pod, it errors out.
Screenshot 2024-09-10 at 8 05 28 PM

**What did you expect:**
Expected `kctrl pkg release` to complete successfully and produce the `carvel-artifacts`.

**Anything else you would like to add:**
[Additional information that will assist in solving the issue.]

**Environment:**

- kapp Controller version (execute `kubectl get deployment -n kapp-controller kapp-controller -o yaml` and the annotation is `kbld.k14s.io/images`):
- Kubernetes version (use `kubectl version`)

---
Vote on this request

This is an invitation to the community to vote on issues, to help us prioritize our backlog. Use the "smiley face" up to the right of this comment to vote.

👍 "I would like to see this addressed as soon as possible"
👎 "There are other more important things to focus on right now"

We are also happy to receive and review Pull Requests if you want to help working on this issue.

### PR DESCRIPTION
**Pull request by devacts**
#### Which issue(s) this PR fixes:

Fixes #1627 

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

##### Review Checklist:

- [ ] Follows the [developer guidelines](https://carvel.dev/shared/docs/latest/development_guidelines/)
- [ ] Relevant tests are added or updated
- [ ] Relevant docs in this repo added or updated
- [ ] Relevant carvel.dev docs added or updated in a separate PR and there's
  a link to that PR
- [ ] Code is at least as readable and maintainable as it was before this
  change

#### Additional documentation e.g., Proposal, usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by rohitagg2020** [comment]
I am thinking if we should add a test case or not (As it has been reported by 2 different users)? We can mock the data in the path `/var/run/secrets/kubernetes.io` and try to run `kctrl pkg release` command and ensure that it completes successfully.

File: cli/pkg/kctrl/cmd/app/release/release_cmd_runner_test.go [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestReleaseCmdRunnerForKappCmd(t *testing.T) {
+	ReleaseCmdRunner := NewReleaseCmdRunner(os.Stdout, false, "", false, nil)
+	cmd := exec.Command("kapp", "deploy", "-f", "-", "-a", "pkg-test", "-y")
+	err := ReleaseCmdRunner.RunWithCancel(cmd, nil)
+	require.NoError(t, err)
```
Comment (100mik) [inline cli/pkg/kctrl/cmd/app/release/release_cmd_runner_test.go:16]: Can we check if Stdout is empty?

Comment (100mik) [reply cli/pkg/kctrl/cmd/app/release/release_cmd_runner_test.go:16]: Or rather pass a buffer instead so that we can be sure that the command was not run

Comment (devacts) [reply cli/pkg/kctrl/cmd/app/release/release_cmd_runner_test.go:16]: @100mik Added the check in this commit: https://github.com/carvel-dev/kapp-controller/pull/1638/commits/8185bfa41f9b7d373fac4511456800ae46ecc9c2

File: cli/pkg/kctrl/cmd/app/release/release_cmd_runner_test.go [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+func TestReleaseCmdRunnerForKappCmd(t *testing.T) {
+	var buf bytes.Buffer
+	ReleaseCmdRunner := NewReleaseCmdRunner(&buf, false, "", false, nil)
+	cmd := exec.Command("kapp", "deploy", "-f", "-", "-a", "pkg-test", "-y")
+	err := ReleaseCmdRunner.RunWithCancel(cmd, nil)
+	require.NoError(t, err)
+	expectedLength := 0
+	if actualLength := buf.Len(); actualLength != expectedLength {
+		t.Errorf("Got Buffer length = %d, Expected empty", actualLength)
```
Comment (100mik) [inline cli/pkg/kctrl/cmd/app/release/release_cmd_runner_test.go]: Let's use require here, and should we point kapp to an actual file?

Comment (devacts) [reply cli/pkg/kctrl/cmd/app/release/release_cmd_runner_test.go]: Done: https://github.com/carvel-dev/kapp-controller/pull/1638/commits/6cb1162994523fdfa9a1ee8bcaef50fd103784b3

**Review by 100mik — APPROVED** [review]

**Pull request merged**